packages/fx-core/tests/core/hooks.test.ts

```diff
@@ -1668,4 +1674,110 @@ describe("Middleware", () => {
       }
     });
   });
+  describe("EnvInfoLoaderMW with MultiEnv enabled", () => {
+    const expectedResult = "ok";
+    const projectPath = "mock/this/does/not/exists";
+
+    sandbox.stub(commonTools, "isMultiEnvEnabled").returns(true);
+
+    // stub environmentManager.loadEnvInfo()
+    const envConfig: EnvConfig = {
+      manifest: {
+        values: {
+          appName: {
+            short: "testApp",
+          },
+        },
+      },
+    };
+    const envProfile = new Map<string, any>();
+    const envInfo = {
+      envName: "test",
+      config: envConfig,
+      profile: envProfile,
+    };
+    sandbox.stub(environmentManager, "loadEnvInfo").returns(Promise.resolve(ok(envInfo)));
+
+    // mock fs.existsSync for EnvInfoLoader
+    const originalPathExists = fs.pathExists;
+    sandbox.stub(fs, "pathExists").callsFake(async (path: string) => {
+      if (path === projectPath) {
+        return true;
+      } else {
+        return originalPathExists(path);
+      }
+    });
+
+    function MockProjectSettingsLoaderMW() {
+      return async (ctx: CoreHookContext, next: NextFunction) => {
+        ctx.projectSettings = {
+          appName: "testApp",
+          version: "1.0",
+          projectId: "abcd",
+          solutionSettings: {
+            name: "fx-solution-azure",
+          },
+          activeEnvironment: "test",
+        };
+        await next();
+      };
+    }
+    async function SolutionContextSpyMW(ctx: CoreHookContext, next: NextFunction) {
+      await next();
+      solutionContext = ctx.solutionContext;
+    }
+
+    // test variables
+    let solutionContext: SolutionContext | undefined;
+    beforeEach(() => {
+      solutionContext = undefined;
+    });
+
+    it("skip statically", async () => {
+      const inputs: Inputs = {
+        platform: Platform.VSCode,
+        projectPath: projectPath,
+      };
+      class MyClass {
+        tools: Tools = new MockTools();
+        async myMethod(inputs: Inputs): Promise<Result<string, FxError>> {
+          return ok(expectedResult);
+        }
+      }
+
+      hooks(MyClass, {
+        myMethod: [MockProjectSettingsLoaderMW(), EnvInfoLoaderMW(true), SolutionContextSpyMW],
+      });
+      const my = new MyClass();
+      const res = await my.myMethod(inputs);
+      assert.isTrue(res.isOk());
+      assert(solutionContext);
+      // envInfo should be set to a default value when envInfo loading is skipped.
+      assert.isTrue(solutionContext?.envInfo.envName === environmentManager.getDefaultEnvName());
```

Comment:
nit: assert.equal would be preferred here, as it provided more info such like expected string and the actual string when the test case failed.

Reply:
It makes sense. Fixed